DataFormats/L1THGCal/interface/HGCalCluster.h

```diff
@@ -1,68 +1,95 @@
 #ifndef DataFormats_L1Trigger_HGCalCluster_h
 #define DataFormats_L1Trigger_HGCalCluster_h
 
-#include "DataFormats/L1Trigger/interface/L1Candidate.h"
 #include "DataFormats/L1Trigger/interface/BXVector.h"
-#include "DataFormats/L1THGCal/interface/ClusterShapes.h"
-
-namespace l1t {
 
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "DataFormats/L1THGCal/interface/ClusterShapes.h"
 
-  class HGCalCluster : public L1Candidate {
-    public:
-      HGCalCluster(){}
-      HGCalCluster( const LorentzVector p4,
-          int pt=0,
-          int eta=0,
-          int phi=0
-          );
-
-      ~HGCalCluster();
+#include "DataFormats/DetId/interface/DetId.h"
 
-      void setHwPtEm  (uint32_t pt)    {hwPtEm_= pt;}
-      void setHwPtHad (uint32_t pt)    {hwPtHad_ = pt;}
-      void setHwSeedPt(uint32_t pt)    {hwSeedPt_ = pt;}
-      void setSubDet  (uint32_t subdet){subDet_ = subdet;}
-      void setLayer   (uint32_t layer) {layer_ = layer;}
-      void setModule  (uint32_t module) {module_ = module;}
-      void setHOverE  (uint32_t hOverE){hOverE_ = hOverE;}
+#include "Math/Vector3D.h"
+#include "TMath.h"
 
-      bool isValid()      const {return true;}
-      uint32_t hwPtEm()   const {return hwPtEm_;}
-      uint32_t hwPtHad()  const {return hwPtHad_;}
-      uint32_t hwSeedPt() const {return hwSeedPt_;}
 
-      uint32_t subDet() const {return subDet_;}
-      uint32_t layer()  const {return layer_;}
-      uint32_t module()  const {return module_;}
+namespace l1t {
 
-      uint32_t hOverE() const {return hOverE_;}
+    class HGCalCluster : public L1Candidate {
+    public:
 
-      ClusterShapes shapes ;
+        /* constructors and destructor */
+        HGCalCluster(){}
+        HGCalCluster( const LorentzVector p4,
+                      int pt,
+                      int eta,
+                      int phi
+        );
+       
+        HGCalCluster( const l1t::HGCalTriggerCell &tc ); 
+
+        ~HGCalCluster();
+
+        /* trigger-cell collection pertinent to the cluster*/
+        BXVector<const l1t::HGCalTriggerCell*>  tcs() const { return tcs_; }        
+
+        /* helpers */
+        bool isPertinent( const l1t::HGCalTriggerCell &tc, double dR ) const;
+        void addTC( const l1t::HGCalTriggerCell &tc );
+
+        /* set info */
+        void setModule  (uint32_t value) { module_   = value; }
+
+        /* get info */
+        bool isValid()      const { return true;  }
+        uint32_t hwPt()     const { return hwPt_; }
+        double mipPt()      const { return mipPt_; }
+        //uint32_t hwSeedPt() const { return hwSeedPt_; }
+        double dist( const l1t::HGCalTriggerCell &tc ) const; /* return distance in 'cm' */
+        
+        ROOT::Math::XYZVector centre() const { return centre_; }
+        ROOT::Math::XYZVector centreNorm() const { return centre_/centre_.z(); }
```

Comment:
1. For consistency with other parts of the code (geometry, trigger cell), it would be better to use `GlobalPoint` for positions
2. Better to return a const reference instead of a copy of the object
3. This doesn't look like the normalized vector, or the norm of the vector. What does 'Norm' means?